.gitmodules

```diff
@@ -0,0 +1,4 @@
+[submodule "gdal-src/source"]
+	path = gdal-src/source
+	url = https://github.com/OSGeo/gdal
+	rev = 654f4907abbbf6bf4226d58a8c067d134eaf3ce9
```

Comment:
gdal has tags for versions, e.g., https://github.com/OSGeo/gdal/tree/v3.8.3

wouldn't it be better to link to a fixed version instead of a cryptic revision number (I have no idea if this is a version or just a snapshot from GDAL's master)?

Reply:
It seems like that this is not possible. Even the `rev` [specifier is not relevant](https://git-scm.com/docs/gitmodules) according to the documentation. That [stackoverflow comment](https://devops.stackexchange.com/questions/5371/where-is-the-git-submodule-revision-actually-stored) says that git stores the rev in the object tree itself.

So I'm not sure how you want to handle that. I've gone with a submodule as that makes it easier to update the code + pull in other changes/patches.

(The commit currently refers to the 3.8.3 release: https://github.com/OSGeo/gdal/releases/tag/v3.8.3, see the commit mentioned there)

Reply:
Maybe we do not need to use git then. They provide bundled dependencies as zip/tar.gz.

Reply:
The problem with not using git would be that it's much harder to update the bundled version (need to download a new tar file instead of just bumping the submodule) and that it would be much harder to pull in additional fixed (need to create your own tar instead of just bumping the submodule).

Reply:
Well, does it make a difference if we pull a subrepo or download a tar?
Moreover, we shouldn't do any fixes to GDAL that are not part of GDAL, I guess. Otherwise, we will ship a non-existing version.

Reply:
For the end user it does not make a difference. For a person that actively works on the code it makes a huge difference. By having a sub repository it's simple to pull in fixes from upstream and in turn also submit fixes to upstream. Both are things that already happened several times while working on that code. 

That written: If you really prefer having a static tar in your gut repository I can certainly change to that, but I highly doubt that this is a good solution.

Reply:
Can we use the tag instead of the commit hash for the submodule?
If not, can we add a comment for the tag that is represented by the commit hash?

Reply:
Neither a tag nor a commit has is meaningful in that file according to the documentation linked above. That information in tracked in git itself. So I do not see adding a comment there, especially as it's likely to become outdated.

Reply:
Well, you don't want to use a release but rather the repo. Then, there is just a commit hash and not a tag name (that would refer to a release). So as a developer, I have to backtrack the commit hash to a tag and then to a GDAL version. If at least there is a comment that tells me "This hash refers to tag 3.8.4" and this was checked at the PR that changed it, it could be helpful.

Reply:
@ChristianBeilschmidt I really feel that there got something lost here, so let me write that out explicitly again:

* There is no `rev` nor a `tag` entry specified for `.gitmodules` files. 
* Information about which commit/tag is referenced for the given submodule is stored directly in git itself (essentially like the content of other files as well).
* (The `rev` specifier will be removed from this file with the next commit, I just wait for a full review of the whole PR instead of trying to address minor feedback again and again)

Therefore could you clarify where exactly you would like to see that comment about which commit hash exactly?

Documentation for submodules: https://git-scm.com/docs/gitmodules

Reply:
I'm not aware of any way to change how this is displayed in the github user interface. If you checkout the repo locally it will correctly show that there is a tag that corresponds to this commit.